alibi/explainers/cfrl_base.py

```diff
@@ -97,6 +96,7 @@ def __init__(self, size: int = 1000) -> None:
         self.len = 0  # current length of the buffer
         self.size = size  # buffer's maximum capacity
         self.batch_size = 0  # batch size (inferred during `append`)
+        self.C: Optional[np.ndarray] = None  # buffer for the conditional tensor
```

Comment:
So `C` has a bit more logic than the other buffers, so seems like `None` is in fact a valid type.

I'm not super comfortable with the current refactor as the buffer declarations are now split into two places. Would be good if you could have a look here too @RobertSamoilescu .

Reply:
Deferring to @RobertSamoilescu on this one.